ports/esp8266/Makefile

```diff
@@ -1,7 +1,20 @@
+# Select the board to build for: if not given on the command line,
+# then default to GENERIC.
+BOARD ?= GENERIC
+
+# If the build directory is not given, make it reflect the board name.
+BUILD ?= build-$(BOARD)
+
+BOARD_DIR ?= boards/$(BOARD)
+ifeq ($(wildcard $(BOARD_DIR)/.),)
+$(error Invalid BOARD specified: $(BOARD_DIR))
+endif
+
 include ../../py/mkenv.mk
 
 # qstr definitions (must come before including py.mk)
```

Comment:
To match other ports, can you also add:

`-include $(BOARD_DIR)/mpconfigboard.mk`

Note the `-` on the front to make it optional.

Although it won't be used for the generic board, it's possible another board might want it (e.g. to reference additional code outside the board dir (e.g. extmod) or to change compiler/linker flags).